psi4/driver/procrouting/proc.py

```diff
@@ -2291,6 +2297,17 @@ def run_scf(name, **kwargs):
     for k, v in scf_wfn.variables().items():
         core.set_variable(k, v)
 
+
+    # optionally get excited states
+    if do_tdscf:
+        excited_states = run_tdscf_energy(name, ref_wfn=scf_wfn)
+    
+        # This way of storing energies and states seems ... not ideal
+        for n, ex_pair in enumerate(excited_states): 
+            ex_en, ex_state = ex_pair
+            scf_wfn.set_variable(f'STATE {n + 1} EXCITATION ENERGY', ex_en) 
+            scf_wfn.set_variable(f'STATE {n + 1} IRREP', ex_state) 
```

Comment:
You could use `set_array_variable` like e.g. in dipole moments?

Reply:
Agreed, that's a better option